openhands/events/stream.py

```diff
@@ -154,8 +154,8 @@ async def async_add_event(self, event: Event, source: EventSource):
         tasks = []
         for key in sorted(self._subscribers.keys()):
             stack = self._subscribers[key]
-            callback = stack[-1]
-            tasks.append(asyncio.create_task(callback(event)))
+            for callback in stack:
+                tasks.append(asyncio.create_task(callback(event)))
```

Comment:
This stack is here for delegate agents, which would subscribe in the order they're created. Not all of them should be invoked for every event, only the latest I think.

Reply:
Ah got it; I've found an alternative solution and we can close this PR

Reply:
Up to you! FWIW I think it would be okay to create its own key, but as you want.

Reply:
Hm, we might need to think about how to handle this appropriately...

Reply:
An alternate solution could be to add an optional param `force_log: bool = False` in `run_controller` (`Openhands/core/main.py`)

We add a subscription in this function; inside the function we can specify

```

Reply:
If that would work - or if setting "debug" in env works - then perhaps an alternative is to update some logging from 'debug' to 'info' in the agent controller? I'm pretty sure it will be useful to have the event itself, once, at info. e.g. maybe [here](https://github.com/All-Hands-AI/OpenHands/blob/4446d3180f4a4dec7b1ef47f0a36d10be8659b15/openhands/controller/agent_controller.py#L244)